api/src/oscc.c

```diff
@@ -1,55 +1,129 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <linux/can.h>
+#include <linux/can/raw.h>
 #include <net/if.h>
 #include <signal.h>
 #include <stdio.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/types.h>
+#include <sys/syscall.h>
 #include <unistd.h>
+#include <poll.h>
 
 #include "oscc.h"
 #include "internal/oscc.h"
 
 
-static int can_socket = -1;
+static int oscc_can_socket = -1;
+static int vehicle_can_socket = -1;
 
 
+oscc_result_t oscc_init()
+{
+    oscc_result_t result = OSCC_ERROR;
+
+    result = oscc_search_can( &auto_init_all_can, true );
+
+    if( result != OSCC_ERROR )
+    {
+        result = register_can_signal();
+    }
+
+    if ( oscc_can_socket > 0 )
```

Comment:
This `if` and `else` are going to step of the value of `result` set above. Maybe we could check that `result` is an expected value before proceeding.